src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    void register_validPasswordUsers_throwsInvalidDataException() {
+        User actual1 = registrationService.register(VALID_PASSWORD_USER_1);
+        User actual2 = registrationService.register(VALID_PASSWORD_USER_2);
+        assertEquals(VALID_PASSWORD_USER_1, actual1);
+        assertEquals(VALID_PASSWORD_USER_2, actual2);
+    }
+
+    @Test
+    void register_nullPasswordUser_notThrowsInvalidDataException() {
+        var invalidDataException = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(NULL_PASSWORD_USER);
+        });
+        assertEquals("User's password is null!", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_invalidAgeUser_notThrowsInvalidDataException() {
+        var invalidDataException = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(INVALID_AGE_USER);
+        });
+        assertEquals("User's is not valid! It's " + INVALID_AGE_USER.getAge() + ", but min "
+                + "age is " + MIN_AGE, invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_validAgeUsers_throwsInvalidDataException() {
+        User actual1 = registrationService.register(VALID_AGE_USER_1);
+        User actual2 = registrationService.register(VALID_AGE_USER_2);
+        assertEquals(VALID_AGE_USER_1, actual1);
+        assertEquals(VALID_AGE_USER_2, actual2);
+    }
+
+    @Test
+    void register_nullLoginUser_notThrowsInvalidDataException() {
+        var invalidDataException = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(NULL_LOGIN_USER);
+        });
+        assertEquals("User's login is null!", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_invalidLoginUser_notThrowsInvalidDataException() {
+        var invalidDataException = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(INVALID_LOGIN_USER);
+        });
+        assertEquals("User's length of login is " + INVALID_LOGIN_USER.getLogin().length()
+                + ". It shouldn't be shorter than 6 characters!",
+                invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_validLoginUsers_throwsInvalidDataException() {
+        User actual1 = registrationService.register(VALID_LOGIN_USER_1);
+        User actual2 = registrationService.register(VALID_LOGIN_USER_2);
+        assertEquals(VALID_LOGIN_USER_1, actual1);
+        assertEquals(VALID_LOGIN_USER_2, actual2);
+    }
+
+    @Test
+    void register_nullUser_notThrowsInvalidDataException() {
+        var invalidDataException = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(NULL_USER);
+        });
+        assertEquals("User is null!", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_sameNameWithUser1_notThrowsInvalidDataException() {
```

Comment:
This test doesn't throw exception?
make test name more understandable. Don't use numbers in the test names 
```suggestion
    void register_usersWithSameName_throwsRegistrationException() {
```